### PR TITLE
New Feature: Move The Language Selector [ en fr ] To The Top-right Corner Of The Page, And Updated The CONTRIBUTING.md And README.md Files With More Contents.

### PR DESCRIPTION
**Pull request by 0xull**
### Move The Language Selector [ en fr ] To The Top-right Corner Of The Page

__Fixes__: Changed the language selector [ en fr ] component to the top-right corner of the page: on the desktop, it's next to the edit button; on the phone that's next to the hamburger menu. This follows the left-to-right design layout that suggests the language selector should be in the top-right corner of the page, that's where people intuitively would look for it. These fixes are for the issue at [#1305](https://github.com/ocaml/ocaml.org/issues/1305).
## Changes Made
Previous: 
ocaml-prev

Recent: 
ocaml-recent

Also in this PR --->

### Updated The CONTRIBUTING.md And README.md Files With More Contents.

__Fixes__: Updated the CONTRIBUTING.md with more contents on: 
- The Issue, new feature and PR templating when making a PR 
- How to make changes to the build folder, ocaml.org, while working on any issue or bug to visually view the changes on the browser and updating the respective folders outside the ocaml.org folder, (it will also be glad to mention that the ocaml.org folder is also listed among the .gitignore files/folder), and 
- The skills required to contribute to the ocaml.org project and the experience level.

This is a PR to the discussion at #1330 

 *Also updated the README.md to direct contributors to the CONTRIBUTING.md file to provide more contents for clarity on the project.*

### CODE REVIEW

**Comment by 0xull** [comment]
This recent PR is in regards to the discussion at the last PR to the `master` branch with @patricoferris, at [#1333](https://github.com/ocaml/ocaml.org/pull/1333).

This current PR passed all the test and has no conflicts with the base branch. @avsm I would love to know your thoughts on this current PR.

 I apologize for having many previous PRs on this. I really want this to be the last PR on this issue.

File: site/css/ocamlorg.css [hunk 1]
```diff
@@ -522,6 +539,12 @@ ul.translations:after { content: "]"; }
     margin-right: 0px;
   }
 }
+@media (min-width: 980px)and (max-width: 1058px) {
+  #searchform, #search {
+    margin-right: 1.8rem;
+    width: 125px !important;
+  }
+}
```
Comment (pitag-ha) [inline site/css/ocamlorg.css]: This fixes a [different problem](https://github.com/ocaml/ocaml.org/issues/1320), doesn't it? For future reference: in general it is a good practice to keep things separate (separate commits; separate PR's). But nice that this is fixed!

**Review by pitag-ha — COMMENTED** [review]
Thanks, this is nice contribution! In addition to what Patrick said, I've also found the following problem on the tutorials page:

Screenshot from 2021-03-31 18-39-32

That's a tough one, but can you try to figure out how to make sure that the language selector doesn't overlap with the search bar?

And one more thing: if you could mention the two issues this PR closes as described [here](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), that would be great.

**Comment by pitag-ha** [comment]
> I apologize for having many previous PRs on this.

No worries. It would make review easier, if you could just update the PR though. You could even force push to your branch. If you don't know and can't find out how to do that, you can ask on irc, discord or similar :)

File: site/css/ocamlorg.css (hunk 1 above)
Comment (0xull) [reply site/css/ocamlorg.css]: Exactly, it fixes the search component breaking out of its alignment in a particular range of screen displays. 

It didn't really occur to me that it should be raised as a separate issue. Thank you @pitag-ha for making it clearer.

**Comment by 0xull** [comment]
> Thanks, this is nice contribution! In addition to what Patrick said, I've also found the following problem on the tutorials page:
> 
> Screenshot from 2021-03-31 18-39-32
> 
> That's a tough one, but can you try to figure out how to make sure that the language selector doesn't overlap with the search bar?
> 
> And one more thing: if you could mention the two issues this PR closes as described [here](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), that would be great.

Currently viewed the `https://ocaml.org/learn/` from my browser, chrome and this is the current status:

ocaml org-learn


And my patch status on the UI - this issue, `http://127.0.0.1:8080/learn/` gives me this

liver-server-learn-page


From the above. this current issue is either fixed or doesn't exist. PLease let me know how I can replicate this issue on my end.

**Comment by 0xull** [comment]
> And one more thing: if you could mention the two issues this PR closes as described here, that would be great.

Sorry, I don't understand the sentence clearly, but I do think I understand the external link to the Github docs. It was so helpful, thank you so much. 

Please explain the sentence clearer.

**Comment by pitag-ha** [comment]
> PLease let me know how I can replicate this issue on my end.

You have to navigate to the tutorials page: when running the page locally, it's `http://127.0.0.1:8080/learn/tutorials/`  

> Please explain the sentence clearer.

I was asking you to to link this PR to the two issues it fixes. That's done by mentioning the issues in the PR description the way described in the link I sent you. If you do that, those issues get closed automatically when this PR is merged. It's nothing important, but helps reduce some work.

If the github docs (the link I sent you) are a bit confusing: about git and github there are a lot of good questions and answers around the web. E.g. [this one](https://stackoverflow.com/a/7189981) might help as well.

Is that clearer? :)

**Comment by pitag-ha** [comment]
Hi @IkehAkinyemi , how is this going? Do you need any help?

**Comment by avsm** [comment]
@IkehAkinyemi thanks for the contribution again.  There are two separate changes in your PR, which was making reviewing them separately complex. In general, try to submit separate PRs for separate changes, so that they can be independently processed.

I have:
- merged your CONTRIBUTIONS.md changes to the main branch of ocaml.org, so that can be considered merged now (you can verify this by reading https://github.com/ocaml/ocaml.org/blob/master/CONTRIBUTING.md)

- the CSS changes can be handled separately now. I have pushed the commit with those changes to staging.ocaml.org, and will then push them to the live site once verified.

Thanks again for the contribution, and don't worry about the Git mixups. Figuring out the right git workflow is challenging for pretty much any developer :-)

**Comment by pitag-ha** [comment]
Is [this](https://github.com/ocaml/ocaml.org/pull/1345#pullrequestreview-625510910) already fixed (the overlap of the language selector with the search bar)? Or do you think that can be done in a separate PR?

**Comment by 0xull** [comment]
Good day @pitag-ha and @avsm, sorry I was away for 6days ago, I had minor dental surgery. I'm back, good and healthier at teeth. :)

 Hello @pitag-ha, regarding [this](https://github.com/ocaml/ocaml.org/pull/1345#pullrequestreview-625510910), I have worked on it, implemented a UI update for it. Once I make a PR, I will tag you to discuss how intuitive the UI changes and behaviour are for the users.

**Comment by 0xull** [comment]
Hello @avsm, thank you. I'm happy to know [that](https://github.com/ocaml/ocaml.org/pull/1345#issuecomment-814144511). Having contributed to Mozilla, and now to Ocaml, my love for Open source has grown bigger. 


I will make sure that the next PRs are made as separate entities. Currently learning more Git concepts to make my workflow on Github better.

I made more CSS updates and I will make PR once I'm done setting up my forked repo to be up-to-date with the ocaml.org repo.

I would really love to say that it has been a great/wonderful experience working on the Ocaml.org project.

**Pull request merged**